camayoc/tests/qcs/test_profiles.py

```diff
@@ -113,31 +168,71 @@ def test_negative_create_missing_data(cleanup):
             b) hosts
             c) credential id's
     :expectedresults: Error is thrown and no new profile is created.
-    :caseautomation: notautomated
+    :caseautomation: automated
     """
-    pass
+    cred = HostCredential(client=shared_client, password=uuid4())
+    cred.create()
+    cleanup.append(cred)
+    profile = NetworkProfile(
+        client=api.Client(response_handler=api.echo_handler),
+        hosts=['localhost'],
+        credential_ids=[cred._id],
+    )
 
+    # remove hosts from fields
+    delattr(profile, data)
+    create_response = profile.create()
+    assert create_response.status_code in range(400, 500)
+    read_response = profile.create()
+    assert read_response.status_code in range(400, 500)
 
-@pytest.mark.skip
-def test_negative_create_invalid_data(cleanup):
-    """Attempt to create a profile missing various data.
+
+@pytest.mark.parametrize('hosts, creds, profname',
+                         [  # bad credential
+                            (['localhost'], [-1], uuid4()),
+                          # bad host name
+                            (['*invalidhostname*'], None, uuid4()),
+                          # bad name
+                            (['localhost'], None, ''),
+                          ],
+                         ids=['bad-credential',
+                              'bad-hostname',
+                              'bad-name'])
+def test_negative_create_invalid_data(
```

Comment:
I think you could use one parameter (`data`) and have a list of dictionary with the data, something like:

```python
@pytest.mark.parametrize('data', (
    {'hosts': ['localhost'], 'credentials': [-1], 'name': uuid4()},
    ...,
    ...,
))
```

And then instantiate a NetworkProfile like so `profile = NetworkProfile(client=..., **data)`.

With that we can know easily what values each field is receiving and if we want to add a new field in the future will be just a matter of updating the data.